src/test/clojure/clara/test_rules.clj

```diff
@@ -907,6 +907,41 @@
            (set batch-inserts)
            (set single-inserts)))))
 
+(def test-accum-needing-token-partitions-correctly-on-fact-binding
+  (let [q (dsl/parse-query []
+                           [[WindSpeed]
+                            [?ts <- (acc/all) :from [Temperature (= ?loc location)]]])
```

Comment:
I suggest making a rule that uses a filter join (and thus exercises AccumulateWithJoinFilterNode) and then doseq'ing over the two rules in a manner similar to the accumulator tests I've added recently.  It is a little extra code, but since the assertions will be the same and the rule structure should be the same the extra mental overhead for someone reading these tests should be fairly minimal. You might need different facts; basically you'd want a field that is ignored in the AccumulateNode rule but used to join in the AccumulateWithJoinFilterNode rule and have that field be equal in both cases.  Having an unused field be equal in the AccumulateNode case shouldn't do any harm there.